specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2022-06-01-preview/ConfidentialWatchlists.json

```diff
@@ -0,0 +1,441 @@
+                "type": "string"
+              }
+            }
+          },
+          "204": {
+            "description": "No Content"
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../common/2.0/types.json#/definitions/CloudError"
+            }
+          }
+        }
+      },
+      "put": {
+        "x-ms-examples": {
+          "Creates or updates a confidential watchlist.": {
+            "$ref": "./examples/confidentialWatchlists/CreateConfidentialWatchlist.json"
+          },
+          "Creates or updates a confidential watchlist and bulk creates watchlist items.": {
+            "$ref": "./examples/confidentialWatchlists/CreateConfidentialWatchlistAndWatchlistItems.json"
+          }
+        },
+        "tags": [
+          "ConfidentialWatchlists"
+        ],
+        "description": "Create or update a confidential Watchlist and its Watchlist Items (bulk creation, e.g. through text/csv content type). To create a Watchlist and its Items, we should call this endpoint with either rawContent or a valid SAR URI and contentType properties. The rawContent is mainly used for small watchlist (content size below 3.8 MB). The SAS URI enables the creation of large watchlist, where the content size can go up to 500 MB. The status of processing such large file can be polled through the URL returned in Azure-AsyncOperation header.",
+        "operationId": "ConfidentialWatchlists_CreateOrUpdate",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "../../../common/2.0/types.json#/parameters/WorkspaceName"
+          },
+          {
+            "$ref": "./common/WatchlistTypes.json#/parameters/WatchlistAlias"
+          },
+          {
+            "$ref": "./common/WatchlistTypes.json#/parameters/Watchlist"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "./common/WatchlistTypes.json#/definitions/Watchlist"
+            }
+          },
+          "201": {
+            "description": "Created. The response includes the Provisioning State and the Azure-AsyncOperation header. To get the progress of the operation, call GET operation on the URL in Azure-AsyncOperation header field.",
+            "schema": {
+              "$ref": "./common/WatchlistTypes.json#/definitions/Watchlist"
+            },
+            "headers": {
+              "Azure-AsyncOperation": {
```

Comment:
Is the PUT async? Then it should follow 201 + provisioningState pattern. [https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/async-api-reference.md#creatingupdating-using-put](https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/async-api-reference.md#creatingupdating-using-put)

Reply:
Done.